### PR TITLE
Enable NTFS and exFAT drivers for all targets

### PR DESCRIPTION
**Pull request by lamauny**

### CODE REVIEW

File: buildroot-external/kernel/v6.6.y/hassos.config [hunk 1]
```diff
@@ -43,6 +43,11 @@ CONFIG_SQUASHFS=y
 CONFIG_SQUASHFS_XATTR=y
 CONFIG_SQUASHFS_LZ4=y
 CONFIG_BTRFS_FS=m
+CONFIG_NTFS_FS=m
+CONFIG_NTFS3_FS=m
```
Comment (sairon) [inline buildroot-external/kernel/v6.6.y/hassos.config:47]:
```suggestion
CONFIG_NTFS3_FS=m
```

`NTFS_FS` is the legacy driver and is fully replaced by `NTFS3_FS` in Linux 6.9+. It's rather pointless to have both options in HAOS, so I'll keep just the modern one.

File: buildroot-external/kernel/v6.6.y/hassos.config [hunk 2]
```diff
@@ -43,6 +43,11 @@ CONFIG_SQUASHFS=y
...
+CONFIG_EXFAT_FS=m
+CONFIG_FUSE_FS=m
+CONFIG_XFS_FS=m
```
Comment (sairon) [inline buildroot-external/kernel/v6.6.y/hassos.config:51]:
```suggestion
```

It's very unlikely that FUSE will be needed for the use case that this PR aims to address - i.e. mounting a disk with less common (in the Linux world) filesystem. 

Similarly XFS, with its deployment rather in the enterprise segment. Which I'd prefer removing for one another reason, the module binary is quite big and space becomes a problem especially on the x86-64 platform.

**Review by sairon — CHANGES_REQUESTED** [review]
I'd prefer to remove some options for the reasons said below. I understand the value of NTFS or exFAT, especially for people who want to share the removable drive with Windows machines, but some of those are rather improbable to be seen and enabling them would just increase the bloat in the system partition.

File: buildroot-external/kernel/v6.6.y/hassos.config (hunk 2 above)
Comment (lamauny) [reply buildroot-external/kernel/v6.6.y/hassos.config:51]: Ok, that sounds ok for me, thanks

**Review by sairon — APPROVED** [review]

**Comment by trevornorcross** [comment]
> The release notes mentions this and it's a little catchy.
> 
> I really wonder what is the meaning of this in real life.
> 
> Does it mean we can do something we couldn't do before? Can we now plug NTFS drives in it and have them being recognized?

I am also trying to understand this.

**Comment by codyrocco** [comment]
great!

**Comment by BasPost** [comment]
> The release notes mentions this and it's a little catchy.
> 
> I really wonder what is the meaning of this in real life.
> 
> Does it mean we can do something we couldn't do before? Can we now plug NTFS drives in it and have them being recognized?

+1 Would like also to know. I just reinstalled my RPi setup tot Supervisor last weekend to have an external NTFS USB disk mounted.

**Comment by sairon** [comment]
> Does it mean we can do something we couldn't do before? Can we now plug NTFS drives in it and have them being recognized?

See #2723. Not all targets had those drivers enabled in the kernel configs, now it's ensured they would be available consistently.

**Pull request merged**